reindexer/complete_reindex/src/complete_reindex.py

```diff
@@ -2,31 +2,80 @@
 
 import os
 
+from boto3.dynamodb.conditions import Attr
 import boto3
-from botocore.exceptions import ClientError
+
 from wellcome_aws_utils import sns_utils
 
 
-def main(event, _ctxt=None, dynamodb_client=None):
-    print(f'event = {event!r}')
-    dynamodb_client = dynamodb_client or boto3.client('dynamodb')
+def _get_item_version(dynamo_item):
+    dynamo_item_version = 1
+
+    if 'version' in dynamo_item:
+        dynamo_item_version = dynamo_item['version']
+
+    return dynamo_item_version
+
+
+def _update_versioned_item(table, dynamo_item, desired_item):
+    desired_dynamo_conditional_update_version = (
+            _get_item_version(dynamo_item) + 1
+    )
+
+    new_dynamo_item = {**dynamo_item, **desired_item}
```

Comment:
This construction is quite unusual – could we just pass in the item we want to PUT?